jdaviz/app.py

```diff
@@ -1069,7 +1069,55 @@ def get_viewer_ids(self, prefix=None):
     def get_viewer_reference_names(self):
         """Return a list of available viewer reference names."""
         # Cannot sort because of None
-        return [self._viewer_item_by_id(vid)['reference'] for vid in self._viewer_store]
+        return [self._viewer_item_by_id(vid).get('reference') for vid in self._viewer_store]
+
+    def get_first_viewer_reference_name(
+            self, require_no_selected_data=False,
+            require_spectrum_viewer=False,
+            require_spectrum_2d_viewer=False,
+            require_table_viewer=False,
+            require_flux_viewer=False,
+            require_image_viewer=False
```

Comment:
if multiple of these are passed as True, then they're not necessarily "required" anymore, since the first one that matches _any_ of the criteria is returned and dependent on the order the logic is hardcoded in the if statements), right?  I'm not sure if that means the logic should be changed, multiple True values should be forbidden, or if the input arguments should have a different prefix though...

Maybe for now we should just make this method hidden so we don't have to give it to much thought and can change it as we use it more in the future?